core_v2/sources/domains.move

```diff
@@ -52,33 +57,28 @@ module aptos_names::domains {
     /// Reverse lookup registry not initialized. `init_reverse_lookup_registry_v1` must be called first
     const EREVERSE_LOOKUP_NOT_INITIALIZED: u64 = 18;
 
-    struct NameRecordKeyV1 has copy, drop, store {
-        subdomain_name: Option<String>,
-        domain_name: String,
+    /// Tokens require a signer to create, so this is the signer for the collection
+    struct CollectionCapabilityV2 has key, drop {
+        capability: SignerCapability,
     }
 
-    struct NameRecordV1 has copy, drop, store {
-        // This is the property version of the NFT that this name record represents.
-        // This is required to tell expired vs current NFTs apart.
-        property_version: u64,
-        // The time, in seconds, when the name is considered expired
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    struct NameRecordV2 has key, drop {
+        domain_name: String,
+        subdomain_name: Option<String>,
         expiration_time_sec: u64,
-        // The address this name is set to point to
         target_address: Option<address>,
-    }
 
-    /// The main registry: keeps a mapping of NameRecordKeyV1 (domain_name & optional subdomain) to NameRecord
-    struct NameRegistryV1 has key, store {
-        // A mapping from domain name to an address
-        registry: Table<NameRecordKeyV1, NameRecordV1>,
+        mutator_ref: token::MutatorRef,
+        burn_ref: token::BurnRef,
     }
 
-    /// The registry for reverse lookups (aka primary names), which maps addresses to their primary name
+    /// The registry for reverse lookups (aka primary names), which maps addresses to their primary name (token address)
```

Comment:
How about we take this chance to change the name to PrimaryName in code haha?

Reply:
I see. I'm fine with reverse registry. Thanks for the context. I think we could potentially support other TLDs like .move in the future. So it might not be a bad idea to keep the name.